### PR TITLE
Dispose queues after consumers - Improve regression tests speed

### PR DESCRIPTION
**Pull request by Martin-Molinero**
#### Description

- Removing `sleep` and disposing queues after consumers. I believe this should also solve the original issue while saving time.
> For 100 regression tests, this reduces execution time by 16 minutes (100 * 10 sec / 60)
- Will first `Join` with 1-second timeout and `Abort` if it failed (tests showed `Abort` was never called). Adding try-catch just for robustness.
- Using `DisposeSafely` versus `Dispose`. 
- Adding check for `_waitHandle` to verify the handle is not already closed.
#### Related Issue

https://github.com/QuantConnect/Lean/issues/831
#### Motivation and Context

Faster is better
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

Executed unit and regressions tests (multiple times) and verified logs
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Util/ParallelRunner.cs [hunk 1]
```diff
@@ -152,19 +152,16 @@ public void Dispose()
         {
             lock (_sync)
             {
-                if (_holdQueue != null) _holdQueue.Dispose();
-                if (_processQueue != null) _processQueue.Dispose();
-
-                // Wait for _holdQueue disposal be completed
-                Thread.Sleep(10000);
-
                 if (_processQueueThread != null && _processQueueThread.IsAlive) _processQueueThread.Abort();
```
Comment (StefanoRaggi) [inline Common/Util/ParallelRunner.cs]: Love the sleep removal 👍
 
While we're touching this `Dispose` method, we could also try to terminate more gracefully, perhaps replacing the `Thread.Abort` with `Thread.Join` or something else?

Comment (Martin-Molinero) [reply Common/Util/ParallelRunner.cs]: Good idea, I'll add a combination of both (`Join` with timeout) 👍

File: Common/Util/ParallelRunner.cs [hunk 2]
```diff
@@ -152,19 +152,29 @@ public void Dispose()
...
+                        Log.Error(err);
+                    }
+                }
 
                 foreach (var worker in _workers)
                 {
                     worker.DisposeSafely();
                 }
 
+                if (_holdQueue != null) _holdQueue.Dispose();
```
Comment (mchandschuh) [inline Common/Util/ParallelRunner.cs]: Please use the `DisposeSafely()` extension method when performing disposal -- helps to manage logging/error handling of exceptions thrown during disposal.

File: Common/Util/ParallelRunner.cs (hunk 1 above)
Comment (mchandschuh) [reply Common/Util/ParallelRunner.cs]: Great ideas guys, love it!

**Pull request merged**